### PR TITLE
track Famous Last words

### PR DESCRIPTION
**Pull request by theunkn0wn1**
This PR tracks the last thing users say in any bot-visible channel. this is primarily to the benefit of `!grab`.

this adds one public method to the `MechaClient`, 
```py 
   def get_last_message(self, channel: str, user: str) -> typing.Optional[str]:...
```

### CODE REVIEW

File: src/mechaclient.py [hunk 1]
```diff
@@ -78,6 +79,7 @@ def __init__(self, *args, mecha_config=None, **kwargs):
         :param message: message body
         :return:
         """
+        await super(MechaClient, self).on_message(channel, user, message)
```
Comment (theunkn0wn1) [inline src/mechaclient.py]:
```suggestion
        await super().on_message(channel, user, message)
```

File: src/features/message_history.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    __slots__ = [
+        "__channel_history"
+    ]
+
+    def __init__(self, nickname, fallback_nicknames=[], username=None, realname=None, eventloop=None,
+                 **kwargs):
+        super().__init__(nickname, fallback_nicknames, username, realname, eventloop, **kwargs)
+        self.__channel_history: typing.Dict[str, typing.Dict[str, str]] = {}
+
+    async def on_message(self, target, by, message):
```
Comment (Eearslya) [inline src/features/message_history.py]: [Style] Typehinting

File: src/mechaclient.py (hunk 1 above)
Comment (Eearslya) [inline src/mechaclient.py]: [Comment, NonBlocking] Inheritance is scary.

**Review by Eearslya — DISMISSED** [review]
One missing typehinting, but in general, LGTM

**Review by Eearslya — APPROVED** [review]

**Pull request merged**